### PR TITLE
Don't copy current line in Each method

### PR DESCRIPTION
**Pull request by msoap**

### CODE REVIEW

File: byline.go [hunk 1]
```diff
@@ -145,17 +145,15 @@ func (lr *Reader) MapStringErr(filterFn func(string) (string, error)) *Reader {
 	})
 }
 
-// Each - processing each line without changing the line
+// Each - processing each line
```
Comment (seh) [inline byline.go]: - [ ] Consider documenting the constraints on the byte slice supplied to the "filterFn" argument:  
  - Will it be overwritten on the next call?  
    I'm not sure if you're reusing the backing array for reading each line.
  - If so, the caller must not retain it, assuming that it's immutable.

File: README.md [hunk 2]
```diff
@@ -185,15 +185,15 @@ An example in which we get odd lines (for `io.Reader` with 10000 lines):
...
-    Benchmark_Each-4                     	   10000	    933023 ns/op	  165426 B/op	   10013 allocs/op
-    Benchmark_EachString-4               	   10000	   1023951 ns/op	  165426 B/op	   10013 allocs/op
-    Benchmark_AWKMode-4                  	    1000	  11854648 ns/op	 3410392 B/op	   55466 allocs/op
+    Benchmark_NativeScannerBytes-4       	   20000	    312502 ns/op	  215080 B/op	      24 allocs/op
+    Benchmark_NativeScannerOnlyCount-4   	   30000	    217491 ns/op	    4160 B/op	       4 allocs/op
+    Benchmark_MapBytes-4                 	   10000	    567421 ns/op	  135184 B/op	      17 allocs/op
+    Benchmark_MapString-4                	    5000	   1408956 ns/op	  374000 B/op	   15018 allocs/op
+    Benchmark_Grep-4                     	   10000	    592100 ns/op	  135200 B/op	      18 allocs/op
+    Benchmark_GrepString-4               	    5000	   1151309 ns/op	  294416 B/op	   10019 allocs/op
+    Benchmark_Each-4                     	   10000	    562337 ns/op	    6201 B/op	      13 allocs/op
```
Comment (seh) [inline README.md]: Hey, a 40% reduction!

Comment (msoap) [inline README.md]: And 10000 -> 13 for allocations :)

File: byline.go (hunk 1 above)
Comment (msoap) [inline byline.go]: Yes, slice can be overwritten in next filters. Small example:
```Go
	reader := strings.NewReader("111\n222\n333")

	i := 0
	var firstLine []byte
	out, _ := byline.NewReader(reader).Each(func(line []byte) {
		i++
		if i == 1 {
			firstLine = line
		}
	}).Each(func(line []byte) {
		line[0] = 'Z'
	}).ReadAllString()

	fmt.Println(string(out))
	fmt.Println("firstLine:", string(firstLine))
```
Added comment about this in d8e92fd28adcac0209fb477b47401093d4f853be

Comment (seh) [inline byline.go]: Presumably that example prints the following:
```
Z11
Z22
Z33
firstLine: Z33
```
Is that correct?

Comment (msoap) [inline byline.go]: No, the first line remains, but may change in the next steps,
result is:
```
Z11
Z22
Z33
firstLine: Z11
```

Comment (seh) [inline byline.go]: Does that mean that it's not the same backing array being overwritten in each step of `Each`? Oh, or is it that the backing array contains the full content, and you're presenting a moving window over it with each callback, where each byte slice is pointing to a different piece of the same backing array (here, the string supplied to `strings.NewReader`)?

Comment (msoap) [inline byline.go]: Under the hood is a slice that returns `scanner.Bytes()`.
See https://github.com/golang/go/blob/master/src/bufio/scan.go#L139 for implementation. This is a bit complicated code :)

**Pull request merged**